curvefs/src/client/s3/client_s3_adaptor.cpp

```diff
@@ -0,0 +1,623 @@
+/*
+ *  Copyright (c) 2020 NetEase Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+/*
+ * Project: curve
+ * Created Date: 21-5-31
+ * Author: huyao
+ */
+
+#include <brpc/channel.h>
+#include <brpc/controller.h>
+
+#include "curvefs/src/client/s3/client_s3_adaptor.h"
+
+namespace curvefs {
+
+namespace client {
+
+void S3ClientAdaptor::Init(const S3ClientAdaptorOption option,
```

Comment:
using ”const S3ClientAdaptorOption &option“ to avoid copying

Reply:
modified